packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -3646,6 +3646,51 @@ describeWithDOM('mount', () => {
         expect(wrapper.prop('bar')).to.equal('bye');
       });
     });
+
+    describe('props in async handler', () => {
+      class TestComponent extends React.Component {
+        constructor(props) {
+          super(props);
+          this.state = { counter: 1 };
+          this.handleClick = this.handleClick.bind(this);
+          this.delay = this.delay.bind(this);
+        }
+
+        async handleClick() {
```

Comment:
we can't actually use async functions in enzyme tests, but in this case it shouldn't matter since this function returns a promise.